src/KafkaFlow.Retry/Durable/Polling/QueueTracker.cs

```diff
@@ -1,130 +1,171 @@
+                    new
+                    {
+                        SchedulerId = this.schedulerId,
+                        PollingJobType = jobDataProvider.PollingDefinition.PollingJobType.ToString(),
+                        TriggerKey = trigger.Key.ToString(),
+                        UnscheduledJob = unscheduledJob.ToString()
+                    });
+            }
+        }
+
+        private void ScheduleJob(IJobDataProvider jobDataProvider, CancellationToken cancellationToken)
+        {
+            try
+            {
+                var job = jobDataProvider.GetPollingJobDetail();
+                var trigger = this.triggerProvider.GetPollingTrigger(this.schedulerId, jobDataProvider.PollingDefinition);
+
                 var scheduledJob = this.scheduler
-                    .ScheduleJob(this.job, this.trigger, cancellationToken)
+                    .ScheduleJob(job, trigger, cancellationToken)
                     .GetAwaiter()
                     .GetResult();
 
                 this.logHandler.Info(
-                    "PollingJob Scheduler Scheduled",
+                    "PollingJob Scheduler scheduled",
                     new
                     {
-                        PollingId = this.retryDurablePollingDefinition.Id,
-                        CronExpression = this.retryDurablePollingDefinition.CronExpression,
+                        SchedulerId = this.schedulerId,
+                        PollingJobType = jobDataProvider.PollingDefinition.PollingJobType.ToString(),
+                        CronExpression = jobDataProvider.PollingDefinition.CronExpression,
                         ScheduleJob = scheduledJob.ToString()
                     });
             }
             catch (Exception ex)
             {
                 this.logHandler.Error(
-                    "PollingJob Scheduler Error",
+                    "PollingJob Scheduler ERROR scheduling",
                     ex,
                     new
                     {
-                        PollingId = this.retryDurablePollingDefinition.Id,
-                        CronExpression = this.retryDurablePollingDefinition.CronExpression
+                        SchedulerId = this.schedulerId,
+                        PollingJobType = jobDataProvider.PollingDefinition.PollingJobType.ToString(),
+                        CronExpression = jobDataProvider.PollingDefinition.CronExpression
                     });
             }
         }
 
-        internal void UnscheduleJob(CancellationToken cancellationToken = default)
+        private void StartScheduler(CancellationToken cancellationToken)
         {
-            this.logHandler.Info(
-                "PollingJob Unscheduler Started",
-                new
-                {
-                    TriggerKey = this.trigger.Key.ToString()
-                });
+            Guard.Argument(this.scheduler).Null(s => "Scheduler was already started. Please call this method just once.");
```

Comment:
Move the guard to inside the lock block, or we might have a race condition here if multiple threads end up here at the same time.